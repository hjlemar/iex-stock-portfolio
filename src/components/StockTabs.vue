<template>
  <div>
    <v-tabs v-model="active" color="indigo" dark >
      <v-tab ripple>Recent</v-tab>
      <v-tab ripple >Portfolio </v-tab>
      <v-tab-item>
        <v-card flat>
          <app-stock-list :stocks='stocks'></app-stock-list>
        </v-card>
      </v-tab-item>
      <v-tab-item>
        <v-card flat>
          <app-porfolio-list :portfolios='portfolios'></app-porfolio-list>
        </v-card>
      </v-tab-item>
    </v-tabs>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import StockList from '@/components/stock/StockList';
import PortfolioList from '@/components/portfolio/PortfolioList';

export default {
  components: {
    appStockList: StockList,
    appPorfolioList: PortfolioList,
  },
  data() {
    return {
      active: null,
      tabs: [
        { title: 'Recent', component: StockList },
        // {title: 'Portfolio', component: StockList}
      ],
    };
  },
  computed: {
    ...mapGetters({
      stocks: 'recentList',
      portfolios: 'getPortfolios',
    }),
  },
};
</script>

<style>

</style>
