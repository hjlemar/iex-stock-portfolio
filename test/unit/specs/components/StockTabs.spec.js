import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import StockTabs from '@/components/StockTabs';
import Vuetify from 'vuetify';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('StockTabs', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({ });
  });

  it('should render component and match snapshot', () => {
    const wrapper = mount(StockTabs, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
