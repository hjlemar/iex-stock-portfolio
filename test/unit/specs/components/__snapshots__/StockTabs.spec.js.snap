// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockTabs should render component and match snapshot 1`] = `
<div>
    <div class="tabs">
        <div class="tabs__bar theme--dark indigo">
            <!---->
            <div class="tabs__wrapper">
                <div class="tabs__container">
                    <div class="tabs__slider-wrapper">
                        <div class="tabs__slider accent"></div>
                    </div>
                    <div class="tabs__div">
                        <a class="tabs__item">Recent</a>
                    </div>
                    <div class="tabs__div">
                        <a class="tabs__item">Portfolio </a>
                    </div>
                </div>
            </div>
            <!---->
        </div>
        <div class="tabs__items">
            <div class="tabs__content" id="undefined" style="display: none;">
                <div class="card card--flat" style="height: auto;">
                    <div class="list list--dense"></div>
                </div>
            </div>
            <div class="tabs__content" id="undefined" style="display: none;">
                <div class="card card--flat" style="height: auto;">
                    <div class="layout row wrap"></div>
                </div>
            </div>
        </div>
    </div>
</div>
`;
